core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -42,6 +42,8 @@ private[spark] abstract class SparkFunSuite extends FunSuite with Logging {
       test()
     } finally {
       logInfo(s"\n\n===== FINISHED $shortSuiteName: '$testName' =====\n")
+      // Avoid leaking map entries in tests that use accumulators without SparkContext
+      Accumulators.clear()
```

Comment:
accumulators are cleaned lazily from GC, so they're usually not cleaned up by the time the test finishes. Actually most of the time this map won't be empty.